### PR TITLE
feat: add text encoding option

### PR DESCRIPTION
**Pull request by tbnritzdoge**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
 		"@typescript-eslint/init-declarations": 0,
 		"no-underscore-dangle": 0,
 		"no-extra-parens": 0,
-		"lines-around-comment": 0
+		"lines-around-comment": 0,
+		"valid-jsdoc": 0
```
Comment (lveneris) [inline .eslintrc.json]: This should be reverted - we had `valid-jsdoc` enabled as it still serves useful for ensuring TypeScript functions are documented properly.

File: src/lib/PetitioResponse.ts [hunk 2]
```diff
@@ -63,9 +63,11 @@ export class PetitioResponse {
 	}
 
 	/**
-	 * @return {*} The response body decoded as a UTF-8 string from the buffer.
+	 * @param {BufferEncoding} [encoding="utf8"] The encoding to use.
+	 * @return {string} The response body decoded as a UTF-8 (or whatever specified in `encoding`) string from the buffer.
+	 * @memberof PetitioResponse
```
Comment (lveneris) [inline src/lib/PetitioResponse.ts]:
```suggestion
```
This is unnecessary, as TypeDoc can figure out which class this function belongs to.

File: src/lib/PetitioResponse.ts [hunk 3]
```diff
@@ -63,9 +63,11 @@ export class PetitioResponse {
 	}
 
 	/**
-	 * @return {*} The response body decoded as a UTF-8 string from the buffer.
+	 * @param {BufferEncoding} [encoding="utf8"] The encoding to use.
+	 * @return {string} The response body decoded as a UTF-8 (or whatever specified in `encoding`) string from the buffer.
```
Comment (lveneris) [inline src/lib/PetitioResponse.ts]:
```suggestion
	 * @return {string} The response body decoded as as a string from the buffer, using either the encoding specified in `encoding` or UTF-8 by default..
```
Just nitpicking for formality

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Aside from these small changes, LGTM.

**Pull request merged**